azerothcore/azerothcore-wotlk
# fix(DB/spell): Feral Spells are not gated at trainers

## Changes Proposed:
-  Made feral spells require their respective forms in order to train.

## Issues Addressed:

- Closes #1666

## SOURCE:


## Tests Performed:

- Created a druid and levelled it to 80 via `.levelup`
- Went to a druid trainer to ensure I could not train any feral abilities without first learning bear or cat form.


## How to Test the Changes:


1.  Create a druid and level it to 80 via `.levelup`
2. Go to a druid trainer (https://wowgaming.altervista.org/aowow/?npc=12042)
3. Feral spells should be unavailable to learn until you train their required shapeshift forms.

## Known Issues and TODO List:


- [Feral Faerie Fire could not be included because it requires Bear OR Cat form, which the spell_required db does not support]


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
